internal/controller/vrg_volsync.go

```diff
@@ -640,13 +673,48 @@ func protectedPVCAnnotations(pvc corev1.PersistentVolumeClaim) map[string]string
 	return res
 }
 
+func (v *VRGInstance) pvcUnprotectVolSyncIfDeleted(
```

Comment:
This function is called from `reconcilePVCAsVolSyncPrimary` which maybe processing prepareFinalSync or runFinalSync, at which point PVC deletion is due to application cleanup and not due to user deletion. A latch like [this](https://github.com/RamenDR/ramen/blob/23ffe92bd32c04646c4102ffb5839e1213402211/internal/controller/vrg_volrep.go#L812-L820) is required here?